src/main/java/fr/xephi/authme/service/SpectateLoginService.java

```diff
@@ -0,0 +1,86 @@
+package fr.xephi.authme.service;
+
+import fr.xephi.authme.settings.properties.RestrictionSettings;
+import org.bukkit.GameMode;
+import org.bukkit.Location;
+import org.bukkit.entity.ArmorStand;
+import org.bukkit.entity.Player;
+
+import javax.inject.Inject;
+import java.util.HashMap;
+import java.util.Map;
+
+public class SpectateLoginService {
+
+    private Map<Player, ArmorStand> armorStands = new HashMap<>();
+    private Map<Player, GameMode> gameModeMap = new HashMap<>();
+
+    @Inject
+    private CommonService service;
+
+    /**
+     * Creates a stand for the player
+     *
+     * @param player the player
+     */
+    public void createStand(Player player) {
+        Location location = player.getLocation();
+        ArmorStand stand = spawnStand(location);
+
+        armorStands.put(player, stand);
+        gameModeMap.put(player, player.getGameMode());
+
+        player.setGameMode(GameMode.SPECTATOR);
+        player.setSpectatorTarget(stand);
+    }
+
+    /**
+     * Updates spectator target for the player
+     *
+     * @param player the player
+     */
+    public void updateTarget(Player player) {
+        ArmorStand stand = armorStands.get(player);
+        if (stand != null) {
+            player.setSpectatorTarget(stand);
+        }
+    }
+
+    /**
+     * Removes the player's stand and deletes effects
+     *
+     * @param player the player
+     */
+    public void removeStand(Player player) {
+        ArmorStand stand = armorStands.get(player);
+        if (stand != null) {
+
+            stand.remove();
+            player.setSpectatorTarget(null);
+            player.setGameMode(gameModeMap.get(player));
+
+            gameModeMap.remove(player);
+            armorStands.remove(player);
+        }
+    }
+
+    public boolean hasStand(Player player) {
+        return armorStands.get(player) != null;
```

Comment:
can be `armorStands.containsKey(player)`

Reply:
Changed to it